File: clearing-house-processors/src/main/java/de/fhg/aisec/ids/clearinghouse/Configuration.java
package de.fhg.aisec.ids.clearinghouse;

import de.fhg.aisec.ids.idscp2.default_drivers.daps.aisec_daps.AisecDapsDriverConfig;
import de.fhg.aisec.ids.idscp2.default_drivers.daps.aisec_daps.SecurityRequirements;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import java.nio.file.Paths;

class Configuration {

    private final static Logger LOG = LoggerFactory.getLogger(Configuration.class);

    private static final String DAPS_ENV_VARIABLE = "TC_DAPS_URL";
    private static final String KEYSTORE_PASS_ENV_VARIABLE = "TC_KEYSTORE_PW";
    private static final String TRUSTSTORE_PASS_ENV_VARIABLE = "TC_TRUSTSTORE_PW";
    // keep this in sync with libraryVersions.yaml
    private static final String INFOMODEL_VERSION = "4.1.0";
    private static final String SENDER_AGENT = "TC_CH_AGENT";
    private static final String ISSUER_CONNECTOR = "TC_CH_ISSUER_CONNECTOR";

    static AisecDapsDriverConfig createDapsConfig(SecurityRequirements securityRequirements){
        var dapsUrl = System.getenv(DAPS_ENV_VARIABLE);
        var keystorePassword = System.getenv(KEYSTORE_PASS_ENV_VARIABLE);
        var truststorePassword = System.getenv(TRUSTSTORE_PASS_ENV_VARIABLE);
        var dapsConfigBuilder = new AisecDapsDriverConfig.Builder()
                .setKeyStorePath(Paths.get("/root/etc/keystore.p12"))
                .setTrustStorePath(Paths.get("/root/etc/truststore.p12"))
                .setKeyAlias("1")
                .setSecurityRequirements(securityRequirements);

        if (dapsUrl != null){
            dapsConfigBuilder.setDapsUrl(dapsUrl);
        }

        if (keystorePassword != null){
            dapsConfigBuilder.setKeyStorePassword(keystorePassword.toCharArray());
        }

        if (truststorePassword != null){
            dapsConfigBuilder.setTrustStorePassword(truststorePassword.toCharArray());
        }

        return dapsConfigBuilder.build();
    }

    static String getInfomodelVersion(){
        return INFOMODEL_VERSION;
    }

    static String getSenderAgent(){
        return getEnvVariable(SENDER_AGENT);
    }

    static String getIssuerConnector(){
        return getEnvVariable(ISSUER_CONNECTOR);
    }

    private static String getEnvVariable(String envVariable){
        var value = System.getenv(envVariable);
        if (value == null) {
            LOG.error("Configuration invalid: Missing {}", envVariable);
            return "";
        }
        else{
            return value;
        }
    }
}

File: clearing-house-processors/src/main/java/de/fhg/aisec/ids/clearinghouse/Configuration.kt
package de.fhg.aisec.ids.clearinghouse

import org.slf4j.LoggerFactory

internal object Configuration {
    private val LOG = LoggerFactory.getLogger(Configuration::class.java)
    private const val LOGGING_SERVICE_ID = "SERVICE_ID_LOG"
    private const val TC_SERVICE_ID = "SERVICE_ID_TC"
    private const val SERVICE_SHARED_SECRET = "SERVICE_SHARED_SECRET"

    val serviceIdTc: String
        get() = getEnvVariable(TC_SERVICE_ID)
    val serviceIdLog: String
        get() = getEnvVariable(LOGGING_SERVICE_ID)
    val serviceSecret: String
        get() = getEnvVariable(SERVICE_SHARED_SECRET)


    private fun getEnvVariable(envVariable: String): String {
        val value = System.getenv(envVariable)
        return if (value == null) {
            LOG.error("Configuration invalid: Missing {}", envVariable)
            ""
        } else {
            value
        }
    }
}
